roles/st2/tasks/datastore.yaml

```diff
@@ -0,0 +1,40 @@
+---
+- name: Add the st2 system service user to the service group
+  become: yes
+  user:
+    user: "{{ st2_system_service_user }}"
+    groups: "{{ st2_system_service_group }}"
+    append: yes
+
+- name: Make stackstorm encryption keys directory
+  become: yes
+  file:
+    path: "{{ st2_datastore_key_file | dirname }}"
+    state: directory
+    mode: 0750
+    owner: root
+    group: "{{ st2_system_service_group }}"
+
+- name: Generate st2 encryption key
+  become: yes
+  command: st2-generate-symmetric-crypto-key --key-path {{ st2_datastore_key_file }}
+  args:
+    creates: "{{ st2_datastore_key_file }}"
+
+- name: Fix permissions on datastore encryption key
+  become: yes
+  file:
+    path: "{{ st2_datastore_key_file }}"
+    mode: 0640
+    owner: root
+    group: st2
+
+- name: Configure encryption key in st2.conf
+  become: yes
+  ini_file:
+    path: /etc/st2/st2.conf
+    section: keyvalue
+    option: encryption_key_path
+    value: "{{ st2_datastore_key_file }}"
+  notify: restart st2api
```

Comment:
@bigmstone Do you remember which service (or services?) is responsible for picking up datastore encryption key?

In our infra I see we just do a `reload`, without service restart.

Reply:
I copied the restart from the st2bootstrap script
```
  sudo st2ctl restart-component st2api
```
https://github.com/StackStorm/st2-packages/blob/master/scripts/st2bootstrap-el6.sh#L492
https://github.com/StackStorm/st2-packages/blob/master/scripts/st2bootstrap-el7.sh#L501
https://github.com/StackStorm/st2-packages/blob/master/scripts/st2bootstrap-deb.sh#L466